great_expectations/core/expectation_suite.py

```diff
@@ -1122,33 +1122,42 @@ class ExpectationSuiteSchema(Schema):
     # NOTE: 20191107 - JPC - we may want to remove clean_empty and update tests to require the other fields;
     # doing so could also allow us not to have to make a copy of data in the pre_dump method.
     # noinspection PyMethodMayBeStatic
-    def clean_empty(self, data):  # noqa: C901
+    def clean_empty(self, data) -> ExpectationSuite | dict:
```

Comment:
```suggestion
    def clean_empty(self, data: T) -> T:
```

If we end up having typing issues here we can use a `T = TypeVar("T", ExpectationSuite, dict)`

Reply:
This is better and I've made this change. Locally mypy was failing because in the `isinstance` block it had didn't infer the correct type (eg ExpectationSuite in that isinstance branch). I will look at this more if it fails on CI.

Reply:
It does fail on CI: https://github.com/great-expectations/great_expectations/actions/runs/8332079731/job/22800389801?pr=9634#step:7:33

Reply:
One last play-by-play post in case you've seen this before. Adding `reveal_type` in the `isinstance(data, ExpectationSuite)` branch returns 2 lines:
```
great_expectations/core/expectation_suite.py:1129: note: Revealed type is "great_expectations.core.expectation_suite.ExpectationSuite"
great_expectations/core/expectation_suite.py:1129: note: Revealed type is "great_expectations.core.expectation_suite.<subclass of "ExpectationSuite" and "dict">"
```

Reply:
The reveal_types do look odd.
I wonder something weird is happening with the `TypeVar` because it's defined as a class constant.
Normally they are defined at the module level.

Reply:
We are hitting https://github.com/python/mypy/issues/10817. This [other bug](https://github.com/python/mypy/issues/14045), which I originally found, is a duplicate of 10817.